2/contrib/s2i/run

```diff
@@ -212,7 +234,27 @@ for FILENAME in ${JENKINS_HOME}/plugins/*.jpi; do
     fi
 done
 
-# for rhel, while the plugins are initially linked to the image volume on initial start,
+# image release migration
+# 1) if the image-version file does not exist, but the configured file does, then we are upgrading from
+# earlier versions of the image which may have not sorted out the migration
+# scenario for PVs; we will do a one time replace of the /var/lib/jenkins/plugins
+# files from the appropriate image source
+if [[ ! -e ${JENKINS_HOME}/image-version && -e ${JENKINS_HOME}/configured ]]; then
+   # 1) if the image-version file does not exist, but the configured file does, then we are upgrading from
+   # earlier versions of the image which may have not sorted out the migration
+   # scenario for PVs; we will do a one time replace of the /var/lib/jenkins/plugins
+   # files from the appropriate image source
+   force_copy_plugins
+elif [ -e ${JENKINS_HOME}/image-version ]; then
+   # 2) if the image-version file does exist, but does not match this image
+   # then do the one time replace
+   currentversion=`cat ${JENKINS_HOME}/image-version`
+   if [ $currentversion != $OPENSHIFT_JENKINS_IMAGE_VERSION ]; then
+      force_copy_plugins
+   fi
+fi
+
+# otherwise, for rhel, while the plugins are initially linked to the image volume on initial start,
 # some maintenance wrt transferring plugins is done on every startup;  this logic
 # covers the case of a release upgrade that introduces new plugins.
```

Comment:
is this actually still necessary then, in light of the other logic added above?

Reply:
To capture new plugins introduced, yes.  My other explanations elsewhere do lend relevant context to this statement.